nfcclient/app.py

```diff
@@ -0,0 +1,83 @@
+import logging
+from time import sleep
+import requests
+
+from config import ClientConfig
+from nfc_reader import NFCReader
+
+logging.basicConfig(
+    format='%(asctime)s %(levelname)-8s %(message)s',
+    level=logging.INFO,
+    datefmt='%Y-%m-%d %H:%M:%S')
+
+try:
+    import RPi.GPIO as GPIO
+except ImportError:
+    print('No GPIO library found')
+    GPIO = None
+
+'''
+NFC readers app
+'''
+
+
+def client_app():
+    client_config = ClientConfig.from_env()
+
+    GPIO.setmode(GPIO.BCM)
+    GPIO.setwarnings(False)
+    GPIO.setup(21, GPIO.OUT)
+    GPIO.output(21, GPIO.LOW)
+
+    readers = []
+    logging.info("Initialise Readers")
+    for door in client_config.doors:
+        for reader in door['readers']:
+            try:
+                nfc_reader = NFCReader(client_config, pin=reader, door=door['name'])
+                readers.append(nfc_reader)
+                logging.info('NFC Reader {} for door {} initialised'.format(reader, door['name']))
+            except Exception as e:
+                logging.critical('NFC Reader {} for door {} failed: {}'.format(reader, door['name'], e))
+
+    logging.info("Start to Listen for cards...")
+    while True:
+        for reader in readers:
+            card = reader.read_card()
+            if card:
+                logging.info('.....CARD Detected.....')
+                card_id = "".join(reader.hex_uid(card))
+                if card_id in client_config.master_keys:
+                    logging.info('Master Card {} Used'.format(card_id))
+                    open_door(reader.door, card_id)
+                    continue
+                elif auth_api_call(client_config, card_id, reader.door):
+                    open_door(reader.door, card_id)
+                    continue
+                else:
+                    logging.warning('Unauthorised Card {}'.format(card_id))
+
+
+def auth_api_call(client_config, card_id, door):
```

Comment:
Why you are sending the whole `client_config` instead of `hub_host` which is the only attribute which is used in the function?